tools/fp20compiler/rc1.0_combiners.cpp

```diff
@@ -49,9 +49,19 @@ void CombinersStruct::Invoke()
             assert(false);
             break;
         }
-        printf("pb_push4f(p, %s, %f, %f, %f, %f);\n", cmd,
-            cc[i].v[0], cc[i].v[1], cc[i].v[2], cc[i].v[3]);
-        printf("p += 5;\n");
+
+        assert(cc[i].v[0] >= 0.0f && cc[i].v[0] <= 1.0f);
+        assert(cc[i].v[1] >= 0.0f && cc[i].v[1] <= 1.0f);
+        assert(cc[i].v[2] >= 0.0f && cc[i].v[2] <= 1.0f);
+        assert(cc[i].v[3] >= 0.0f && cc[i].v[3] <= 1.0f);
+
+        printf("pb_push1(p, %s,", cmd);
+        printf("\n    MASK(0xFF000000, 0x%02X)", (unsigned char)(cc[i].v[3] * 0xFF));
+        printf("\n    | MASK(0x00FF0000, 0x%02X)", (unsigned char)(cc[i].v[0] * 0xFF));
+        printf("\n    | MASK(0x0000FF00, 0x%02X)", (unsigned char)(cc[i].v[1] * 0xFF));
+        printf("\n    | MASK(0x000000FF, 0x%02X)", (unsigned char)(cc[i].v[2] * 0xFF));
+        printf(");\n");
+        printf("p += 2;\n");

```

Comment:
WTF.. `NV097_SET_SPECULAR_FOG_FACTOR` could be float? There's a gap of `2x 4*sizeof(float)`

Reply:
No.